product_operating_unit/models/product_template.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2019 brain-tec AG (http://www.braintec-group.com)
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    operating_unit_ids = fields.Many2many(
+        "operating.unit",
+        "product_operating_unit_rel",
+        string="Operating Units",
+        compute="_compute_operating_unit_ids",
+        store=True,
+        readonly=False,
+        default=lambda self: self._default_operating_unit_ids(),
+    )
+    categ_id = fields.Many2one(
+        default=lambda self: self._get_default_category_id(),
+    )
+
+    @api.model
+    def _default_operating_unit_ids(self):
+        if self.categ_id and self.categ_id.operating_unit_ids:
+            return [(6, 0, self.categ_id.operating_unit_ids.ids)]
```

Comment:
Since this method is used as a default method, `self.categ_id` will always be empty.

Reply:
For now, I simply removed the if statement but I'm wondering if the default is still needed with the compute based on the categ_id